File: examples/node/pingpong.js
var faye = require('../../build/node/faye-node');

ENDPOINT = 'http://localhost:8000/bayeux';
console.log('Connecting to ' + ENDPOINT);

var ping = new faye.Client(ENDPOINT);
ping.subscribe('/ping', function() {
  console.log('PING');
  setTimeout(function() { ping.publish('/pong', {}) }, 1000);
});

var pong = new faye.Client(ENDPOINT);
pong.subscribe('/pong', function() {
  console.log('PONG');
  setTimeout(function() { pong.publish('/ping', {}) }, 1000);
});

setTimeout(function() { ping.publish('/pong', {}) }, 500);

File: examples/ruby/pingpong.rb
require 'rubygems'
require 'bundler/setup'
require 'faye'

EM.run {
  ENDPOINT = 'http://localhost:9292/bayeux'
  puts 'Connecting to ' + ENDPOINT

  ping = Faye::Client.new(ENDPOINT)
  ping.subscribe('/ping') do
    puts 'PING'
    EM.add_timer(1) { ping.publish('/pong', {}) }
  end

  pong = Faye::Client.new(ENDPOINT)
  pong.subscribe('/pong') do
    puts 'PONG'
    EM.add_timer(1) { ping.publish('/ping', {}) }
  end

  EM.add_timer(0.5) { ping.publish('/pong', {}) }
}
